### DISCUSSION TITLE
Convert PNG and Visio files to draw.io XML

### DISCUSSION THREAD

**Discussion by AngusParsonson** [jgraph/drawio#4251] [Q&A]
Hi there,

I am trying to use draw.io in [embed mode](https://www.drawio.com/doc/faq/embed-mode). The iframe takes an XML parameter to preload the editor with a diagram. I want to be able to load PNG and Visio files, so I need some way to convert these file types to valid draw.io mxgraph XML programmatically.

Has anyone encountered this before, or have any ideas about how I could solve it?

Thanks.

**Comment by tiktaa** [comment]
![prepro](https://github.com/user-attachments/assets/3b47bdae-446e-4fb2-a262-07727b242327)
![prepro](https://github.com/user-attachments/assets/7fb06486-b2dc-4ebe-8273-cc58074ef9a5)
